recipes/boost-ext-ut/all/conanfile.py

```diff
@@ -59,45 +67,41 @@ def validate(self):
                         self.settings.compiler.version))
 
     def config_options(self):
-        if tools.Version(self.version) <= "1.1.8":
+        if Version(self.version) <= "1.1.8":
             del self.options.disable_module
 
     def configure(self):
-        if self.settings.compiler in ["msvc", "Visual Studio"]:
-            if "disable_module" in self.options.values:
-                self.options.disable_module = True
+        if is_msvc(self) and "disable_module" in self.options.values:
+            self.options.disable_module = True
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["BOOST_UT_BUILD_BENCHMARKS"] = False
-        self._cmake.definitions["BOOST_UT_BUILD_EXAMPLES"] = False
-        self._cmake.definitions["BOOST_UT_BUILD_TESTS"] = False
-        self._cmake.definitions["PROJECT_DISABLE_VERSION_SUFFIX"] = True
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BOOST_UT_BUILD_BENCHMARKS"] = False
+        tc.cache_variables["BOOST_UT_BUILD_EXAMPLES"] = False
+        tc.cache_variables["BOOST_UT_BUILD_TESTS"] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
```

Comment:
Is it `self.conf` or `self.conf_info`? Documentation about this new feature is scattered across different sections and not very clear.

Reply:
I was mostly just following Conan Center's documentation on the matter: https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md#options. But, testing it out, `self.conf` has to be used in the `generate` method otherwise nothing happens. `self.conf_info` must just be for use in the `package_info` method and perhaps the `validate` method.

Reply:
conf_info can be used in generate() too: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#removed-self-user-info

Reply:
Really? I tested it and adding `-c tools.build:skip_test=False` doesn't actually run the tests when I use `conf_info`, only when I use `conf`.

Reply:
No, I mean, conf_info can be used, but, for user configuration.